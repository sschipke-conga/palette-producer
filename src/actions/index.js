

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setAllProjects = allProjects => ({
  type: 'SET_PROJECTS',
  allProjects
});

export const setCurrentPalette = palette => ({
  type: 'SET_CURRENT_PALETTE',
  palette
});

export const setCurrentPaletteInfo = info => ({
  type: 'SET_CURRENT_PALETTE_INFO',
  info
});

export const setCurrentProjectInfo = info => ({
  type: 'SET_CURRENT_PROJECT_INFO',
  info
});

export const setAllPalettes = allPalettes => ({
  type: 'SET_PALETTES',
  allPalettes
});

export const toggleMenu = () => ({
  type: 'TOGGLE_MENU'
})

export const removePalette = (id) => ({
  type: 'REMOVE_PALETTE',
  id: parseInt(id)
});