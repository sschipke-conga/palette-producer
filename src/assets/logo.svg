<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="316.9996" height="316.9996" viewBox="0 0 316.9996 316.9996"><defs><style>.cls-1{fill:#fff;}.cls-1,.cls-2,.cls-3,.cls-4,.cls-5{stroke:#000;stroke-miterlimit:10;stroke-width:10px;}.cls-2{fill:#fbb03b;}.cls-3{fill:#f15a24;}.cls-4{fill:#d4145a;}.cls-5{fill:#3c4892;}</style></defs><title>Artboard 5</title><path class="cls-1" d="M31.62037,143.03632c.33314-55.44444,42.19532-108.237,95.8371-120.86067C185.4136,8.53673,248.78037,35.47176,273.57154,85.89533c31.15582,63.36876,1.52388,129.99623-69.57942,114.45468-13.68126-2.99042-12.8967-5.45845-31.15819-10.9585-23.80811-7.1706-24.20607,3.48171-14.73851,56.5,5,28,7,39-8,49C111.92072,320.34136,31.22091,209.51685,31.62037,143.03632Z"/><circle class="cls-2" cx="75.01795" cy="138.12396" r="22.92254"/><circle class="cls-3" cx="116.7926" cy="76.84227" r="22.92254"/><circle class="cls-4" cx="195.10246" cy="76.8141" r="22.92254"/><circle class="cls-5" cx="230.17288" cy="147.88452" r="22.92254"/></svg>