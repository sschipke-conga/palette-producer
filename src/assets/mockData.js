export const mockPalettes = [
  {
    id: 30,
    project_id: 35,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z",
  },
  {
    id: 33,
    project_id: 35,
    name: "New one now",
    color1: "#f36d6b",
    color2: "#72cc20",
    color3: "#f11a21",
    color4: "#8eb4e8",
    color5: "#b08980",
    created_at: "2020-02-15T00:06:46.617Z",
    updated_at: "2020-02-15T00:06:46.617Z",
  },
  {
    id: 40,
    project_id: 35,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z",
  },
  {
    id: 45,
    project_id: 37,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z"
  },
  {
    id: 43,
    project_id: 37,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z"
  }
]

export const mockProjects = [
  {
    id: 30,
    name: "My First Project",
  },
  {
    id: 33,
    name: "Second Projects",
  },
  {
    id: 40,
    name: "Third Project"
  }
]

export const mockUser = {
  user_id: 34,
  username: 'Gary'
}

export const mockCurrentPalette = [
  {
    name: 'color1',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color2',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color3',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color4',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color5',
    hexCode: '#FFFFFF',
    isLocked: false
  }
]