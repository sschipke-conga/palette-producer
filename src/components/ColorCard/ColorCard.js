import './ColorCard.scss';
import React from 'react';
import { ChromePicker } from 'react-color';
import PropTypes from 'prop-types'
import { IoIosLock, IoIosUnlock } from 'react-icons/io';

export const ColorCard = ({ index, color, isLocked, changeColor, toggleLock }) => {

  const colorStyles = {
    background: color,
  }

  return (
    <div className='ColorCard'>
      <div className='color' style={colorStyles}>
        {isLocked ?
          <IoIosLock className='lock' onClick={() => toggleLock(index, color)} /> :
          <IoIosUnlock className='lock' onClick={() => toggleLock(index, color)} />}
        <p className='hex'>{color}</p>
      </div>
      <div className='picker-container'>
        <ChromePicker className='color-picker' disableAlpha={true} width={170} color={color} onChange={({ hex }) => changeColor(index, hex)} />
      </div>
    </div >
  )
}

export default ColorCard;

ColorCard.propTypes = {

}
