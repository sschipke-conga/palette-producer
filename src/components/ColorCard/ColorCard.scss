.ColorCard {
  display: flex;
  flex-direction: column;
  width: 20%;
}

.color {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 30vh;
  padding: 20px;
}

.lock {
  cursor: pointer;
  background-color: rgba(244, 244, 244, 0.3);
  border-radius: 5px;
  height: 50px;
  margin: 0 auto;
  width: 50px;
}

.hex {
  bottom: 0px;
  color: whitesmoke;
  font-size: 20px;
  font-weight: 500;
  left: 50%; 
  position: absolute;
  text-shadow: 0.5px 0.5px rgb(113, 113, 113);
  text-transform: uppercase;
  transform: translate(-50%);
}

.picker-container {
  display: flex;
  justify-content: center;
  margin-top: 5px;
  width: 100%;
  min-height: 12.5em;
}

.color-picker {
  width: 90%
}