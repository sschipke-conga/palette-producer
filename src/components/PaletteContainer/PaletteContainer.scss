.PaleteContainer-header {
  display: flex;
  justify-content: space-between;
  padding: 20px 20px;
}

.PaleteContainer-names { 
  width: 40%;
}

.project-name,
.palette-name {
  border-radius: 5px;
  margin: 0px;
  padding: 10px;
  text-align: left;
  width: 100%;
  &:focus { 
    background: #ededed;
    outline: none;
  }
}

.icon-container {
  align-items: center;
  display: flex;
  cursor: pointer;
  margin-right: 30px;
}

.save-icon {
  height: 50px;
  width: 50px;
}

.PaletteContainer-section {
  display: flex;
  justify-content: space-around;
}