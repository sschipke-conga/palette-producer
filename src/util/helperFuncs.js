export const generateRandomHex = () => {
  let hexCode = "#" + Math.floor(Math.random() * 16777215).toString(16);
  if (hexCode.length < 7) {
    return generateRandomHex();
  }
  return hexCode;
};
