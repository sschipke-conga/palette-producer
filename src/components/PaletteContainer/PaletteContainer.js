import React, { Component } from 'react';
import './PaletteContainer.scss';
import PropTypes from 'prop-types';
import { IoIosRefresh } from "react-icons/io";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import {generateRandomHex} from '../../util/helperFuncs'
import {setCurrentPalette} from '../../actions/index'
import { Link } from "react-router-dom";
import PaletteForm from '../PaletteForm/PaletteForm'
import ColorCard from '../ColorCard/ColorCard';

export class PaletteContainer extends Component {
  randomizePalette = () => {
    const {setCurrentPalette, currentPalette} = this.props;
    const updatedPalette = currentPalette.map(item => {
      if(!item.isLocked) {
        item.hexCode = generateRandomHex()
      }
      return item
    })
    setCurrentPalette(updatedPalette)
  }

  displayPalette = () => {
    const {currentPalette} = this.props
    return currentPalette.map((color, index) => {
      return (
        <ColorCard
          key={index}
          index={index}
          color={color.hexCode}
          isLocked={color.isLocked}
        />
      )
    })
  }

  render() {
    const {user} = this.props
    return (
      <main className="PaletteContainer">
        <section className="PaletteContainer-section">
          {this.displayPalette()}
        </section>
        <div className="icon-container">
          <IoIosRefresh
            className="randomize-icon"
            onClick={this.randomizePalette}
          />
        </div>
        {user && <PaletteForm />}
        {!user && (
          <h4 className="login-signup-message">
            Please
            <Link to="/login">Login</Link>
            or 
            <Link to="/signup">Sign up</Link>
            to get started!
          </h4>
        )}
      </main>
    );
  }
}



export const mapStateToProps = state => ({
  currentPalette: state.currentPalette,
  user: state.user,
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setCurrentPalette
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(PaletteContainer);
PaletteContainer.propTypes = {
  user: PropTypes.object.isRequired,
  currentPalette: PropTypes.array.isRequired,
  setCurrentPalette: PropTypes.func.isRequired
}






