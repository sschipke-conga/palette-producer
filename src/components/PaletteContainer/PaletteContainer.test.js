import React from "react";
import { PaletteContainer, mapStateToProps, mapDispatchToProps } from './PaletteContainer';
import {ColorCard} from '../ColorCard/ColorCard'
import { shallow } from "enzyme";
import {mockCurrentPalette} from '../../assets/mockData'
import {generateRandomHex} from '../../util/helperFuncs'
import {setCurrentPalette} from '../../actions'

jest.mock('../../util/helperFuncs')

const mockSetCurrentPalette = jest.fn()

describe('PaletteContainer', () => {
  const mockUser = {name: 'Shanda', id:5};
  let wrapper, colorCards;
  beforeEach(() => {
    // colorCards = shallow(<ColorCard
    //   index={3}
    //   color={'#FFFFFF'}
    //   changeColor={jest.fn()}
    // />)
    wrapper = shallow(<PaletteContainer
      user={mockUser}
      setCurrentPalette={mockSetCurrentPalette}
      currentPalette={mockCurrentPalette}
    />)
  })
  describe("randomizePalette", () => {
    it('should call generateRandomHex', () => {
      wrapper.instance().randomizePalette()
      expect(generateRandomHex).toHaveBeenCalled()
    })
    it('should call setCurrentPalette', () => {
      wrapper.instance().randomizePalette()
      expect(mockSetCurrentPalette).toHaveBeenCalled()
    })
  });
  describe('displayPalette', () => {
    it('should return and array of ColorCards', () => {
      expect(wrapper.instance().displayPalette().length)
      .toEqual(5);
    })
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  describe('mapStateToProps/mapDispatchToProps', () => {
    it('mapStateToProps the user and the state of the menu', () => {
      const mockUser = { username: 'Dave', id: 4 }
      const mockState = {
        currentPalette: mockCurrentPalette,
        user: mockUser,
      };
      const expected = {
        currentPalette: mockCurrentPalette,
        user: mockUser,
      };
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
    it('calls dispatch with setCurrentPalette action when it is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setCurrentPalette('SET_CURRENT_PALETTE', mockCurrentPalette);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCurrentPalette('SET_CURRENT_PALETTE', mockCurrentPalette);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
})
