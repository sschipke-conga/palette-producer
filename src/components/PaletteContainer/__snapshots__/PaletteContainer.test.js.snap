// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteContainer should match the snapshot 1`] = `
<main
  className="PaletteContainer"
>
  <section
    className="PaletteContainer-section"
  >
    <Connect(ColorCard)
      index={0}
      isLocked={false}
      key="0"
    />
    <Connect(ColorCard)
      index={1}
      isLocked={false}
      key="1"
    />
    <Connect(ColorCard)
      index={2}
      isLocked={false}
      key="2"
    />
    <Connect(ColorCard)
      index={3}
      isLocked={false}
      key="3"
    />
    <Connect(ColorCard)
      index={4}
      isLocked={false}
      key="4"
    />
  </section>
  <div
    className="icon-container"
  >
    <IoIosRefresh
      className="randomize-icon"
      onClick={[Function]}
    />
  </div>
  <Connect(PaletteForm) />
</main>
`;
